path in HF_ENDPOINT discarded in v0.28.0

### Describe the bug

Hi,
We are using Artifactory to proxy the download of models from HuggingFace.
We set the `HF_ENDPOINT` env var to `https://<artifactory-host>/artifactory/api/huggingfaceml/huggingface-ml-external` for the lib to connect directly to Artifactory.

When using this config in the version 0.27.1, we can see in the lib debug logs that the correct endpoint is used to download
```py
from huggingface_hub import snapshot_download

snapshot_download(
    repo_id="meta-llama/Llama-3.1-8B-Instruct", revision="main"
)

```

```
Request dc69bb46-b7d7-40d4-9b7e-619916bb614a: HEAD https://<artifactory-host>/artifactory/api/huggingfaceml/huggingface-ml-external/meta-llama/Llama-3.1-8B-Instruct/resolve/be9e14a5ab73b04d65b74f7dd64e44cc7f77d5cf/model-00002-of-00004.safetensors (authenticated: False)
Request 4117cde9-71fa-423d-9056-3c8a92739623: HEAD https://<artifactory-host>/artifactory/api/huggingfaceml/huggingface-ml-external/meta-llama/Llama-3.1-8B-Instruct/resolve/be9e14a5ab73b04d65b74f7dd64e44cc7f77d5cf/model-00004-of-00004.safetensors (authenticated: False)
Request 4eff2085-7412-4081-89cc-0bc858088423: HEAD https://<artifactory-host>/artifactory/api/huggingfaceml/huggingface-ml-external/meta-llama/Llama-3.1-8B-Instruct/resolve/be9e14a5ab73b04d65b74f7dd64e44cc7f77d5cf/model-00001-of-00004.safetensors (authenticated: False)
Request 09ed572a-6ea4-4922-8385-be611dd7f046: HEAD https://<artifactory-host>/artifactory/api/huggingfaceml/huggingface-ml-external/meta-llama/Llama-3.1-8B-Instruct/resolve/be9e14a5ab73b04d65b74f7dd64e44cc7f77d5cf/original/consolidated.00.pth (authenticated: False)
```

However, since version `0.28.0`, the needed path `/artifactory/api/huggingfaceml/huggingface-ml-external/` in our request URL is removed, which results in a 404


```
Request 83c6aab3-961b-4d67-aa75-e48b5b492143: HEAD https://<artifactory-host>/meta-llama/Llama-3.1-8B-Instruct/resolve/be9e14a5ab73b04d65b74f7dd64e44cc7f77d5cf/model-00002-of-00004.safetensors (authenticated: False)
Request 5a096eec-300b-4cac-98c6-43b6802df61c: HEAD https://<artifactory-host>/meta-llama/Llama-3.1-8B-Instruct/resolve/be9e14a5ab73b04d65b74f7dd64e44cc7f77d5cf/model-00001-of-00004.safetensors (authenticated: False)
Request d2ebada7-233e-479f-83ba-6cf742a58458: HEAD https://<artifactory-host>/meta-llama/Llama-3.1-8B-Instruct/resolve/be9e14a5ab73b04d65b74f7dd64e44cc7f77d5cf/model-00003-of-00004.safetensors (authenticated: False)
[...]
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://<artifactory-host>/meta-llama/Llama-3.1-8B-Instruct/resolve/be9e14a5ab73b04d65b74f7dd64e44cc7f77d5cf/model-00001-of-00004.safetensors
```

I think this difference comes from this change: https://github.com/huggingface/huggingface_hub/commit/438f2fbfb9785493a879316cdeed8f36f793cd98#diff-b9ea02465324089a58bcb914a78b6c50143dfa0aadf772cef27478581f1346bcR71
Where instead of concatenating the model path to the `HF_ENDPOINT` var, it uses `urljoin`. which as I tested, removes the path after the hostname

```py
from urllib.parse import urljoin

HF_ENDPOINT="https://artifactory-host/artifactory/api/huggingfaceml/huggingface-ml-external"
ENDPOINT = HF_ENDPOINT.rstrip("/")
HUGGINGFACE_CO_URL_TEMPLATE = urljoin(ENDPOINT, "/{repo_id}/resolve/{revision}/{filename}")

print(HUGGINGFACE_CO_URL_TEMPLATE)
```
Result: `https://artifactory-host/{repo_id}/resolve/{revision}/{filename}`

Would it be possible to revert this change, or to make it work with `HF_ENDPOINT` endpoints that contain path ?

Thank you in advance

### Reproduction

Described above

### Logs

```shell
Logs above
```

### System info

```shell
Request 9e55649d-a4ca-4997-b5a6-f83642e9abd0: GET https://artifactory-host/artifactory/api/huggingfaceml/huggingface-ml-external/api/whoami-v2 (authenticated: False)

Copy-and-paste the text below in your GitHub issue.

- huggingface_hub version: 0.28.0
- Platform: Windows-10-10.0.19045-SP0
- Python version: 3.11.9
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Running in Google Colab Enterprise ?: No
- Token path ?: C:\Users\<user>\.cache\huggingface\token
- Has saved token ?: True
- Configured git credential helpers:
- FastAI: N/A
- Tensorflow: N/A
- Torch: N/A
- Jinja2: 3.1.4
- Graphviz: N/A
- keras: N/A
- Pydot: N/A
- Pillow: N/A
- hf_transfer: N/A
- gradio: N/A
- tensorboard: N/A
- numpy: 2.2.2
- pydantic: N/A
- aiohttp: N/A
- ENDPOINT: https://artifactory-host/artifactory/api/huggingfaceml/huggingface-ml-external
- HF_HUB_CACHE: C:\Users\<user>\.cache\huggingface\hub
- HF_ASSETS_CACHE: C:\Users\<user>\.cache\huggingface\assets
- HF_TOKEN_PATH: C:\Users\<user>\.cache\huggingface\token
- HF_STORED_TOKENS_PATH: C:\Users\<user>\.cache\huggingface\stored_tokens
- HF_HUB_OFFLINE: False
- HF_HUB_DISABLE_TELEMETRY: False
- HF_HUB_DISABLE_PROGRESS_BARS: None
- HF_HUB_DISABLE_SYMLINKS_WARNING: False
- HF_HUB_DISABLE_EXPERIMENTAL_WARNING: False
- HF_HUB_DISABLE_IMPLICIT_TOKEN: False
- HF_HUB_ENABLE_HF_TRANSFER: False
- HF_HUB_ETAG_TIMEOUT: 86400
- HF_HUB_DOWNLOAD_TIMEOUT: 86400
```

Hello @Wauplin, thanks for the fast reply, I can do a PR.
However by adding an f-string the the constant.py file, I get an error when running the tests
```
HUGGINGFACE_CO_URL_TEMPLATE = f"{ENDPOINT}/{repo_id}/resolve/{revision}/{filename}"
E   NameError: name 'repo_id' is not defined
```

Indeed, the path in templated in https://github.com/huggingface/huggingface_hub/blob/ef46eae16a761c51408974ffe17c17caf81008b9/src/huggingface_hub/file_download.py#L249

I think that's why before using `urljoin` string concat was used
`HUGGINGFACE_CO_URL_TEMPLATE = ENDPOINT + "/{repo_id}/resolve/{revision}/{filename}"`

Should I just replace the two urljoin with string concat again ?

--- a/src/huggingface_hub/constants.py
+++ b/src/huggingface_hub/constants.py
@@ -2,7 +2,6 @@ import os
 import re
 import typing
 from typing import Literal, Optional, Tuple
-from urllib.parse import urljoin
 
 
 # Possible values for env variables
@@ -68,7 +67,7 @@ ENDPOINT = os.getenv("HF_ENDPOINT", "").rstrip("/") or (
     _HF_DEFAULT_STAGING_ENDPOINT if _staging_mode else _HF_DEFAULT_ENDPOINT
 )
 
-HUGGINGFACE_CO_URL_TEMPLATE = urljoin(ENDPOINT, "/{repo_id}/resolve/{revision}/{filename}")
+HUGGINGFACE_CO_URL_TEMPLATE = ENDPOINT + "/{repo_id}/resolve/{revision}/{filename}"
 HUGGINGFACE_HEADER_X_REPO_COMMIT = "X-Repo-Commit"
 HUGGINGFACE_HEADER_X_LINKED_ETAG = "X-Linked-Etag"
 HUGGINGFACE_HEADER_X_LINKED_SIZE = "X-Linked-Size"
@@ -79,7 +78,7 @@ INFERENCE_ENDPOINT = os.environ.get("HF_INFERENCE_ENDPOINT", "https://api-infere
 INFERENCE_ENDPOINTS_ENDPOINT = "https://api.endpoints.huggingface.cloud/v2"
 
 # Proxy for third-party providers
-INFERENCE_PROXY_TEMPLATE = urljoin(ENDPOINT, "/api/inference-proxy/{provider}")
+INFERENCE_PROXY_TEMPLATE = ENDPOINT + "/api/inference-proxy/{provider}"
 
 REPO_ID_SEPARATOR = "--"
 # ^ this substring is not allowed in repo_ids on hf.co

